### PR TITLE
Wait for zone and record set promises in portal before displaying record sets

### LINKED ISSUE #141: Records with same name as zone sometimes show up as a dotted host (invalid)
**Describe the bug**
Sometimes, when loading the main zone page with record sets, records with the same name as the zone (which is valid) get flagged as a dotted host.

**VinylDNS Version**
Latest

**Additional context**
In the records controller, we call both getZone and listRecordSets on page load. After listRecordSets returns, we determine if each recordset is dotted based on comparison with the zone name. If getZone returns last, this comparison fails for records that are the same as the zone name, resulting in the dotted host warning when it should not.

We probably still want these calls to happen at the same time, but need to wait to do the comparison until both have completed.

### PR DESCRIPTION
**Pull request by mitruly**
Need to wait for promises from multiple asynchronous calls to return before displaying records so that dotted host warning message does not erroneously appear.

Fixes #141.

Changes in this pull request:
- Update `controller.records.js` so that calls to `toDisplayRecord` wait for resolved zone and records promises.
- Update `batch-change-new-controller.js` and batch change view to remove unnecessary `event.preventDefault()` calls.

----
Note to reviewers: 
- Tested this by increasing the timeout of `loadZonesPromise = $timeout($scope.refreshZone, 0);` from `0` to `2000`.
- Information about the `$q` service/promise construction ca be found at: https://docs.angularjs.org/api/ng/service/$q

### CODE REVIEW

File: modules/portal/public/lib/controllers/controller.records.js [hunk 1]
```diff
@@ -391,17 +394,20 @@ angular.module('controller.records', [])
...
-            newRecords.push(recordsService.toDisplayRecord(record, $scope.zoneInfo.name));
-        });
-        $scope.records = newRecords;
-        if($scope.records.length > 0) {
-          $("td.dataTables_empty").hide();
-        } else {
-          $("td.dataTables_empty").show();
-        }
-    }
+        $q.all([loadZonesPromise, loadRecordsPromise])
```
Comment (rebstar6) [inline modules/portal/public/lib/controllers/controller.records.js]: did you look into testing this in controller.records.spec.js?

**Comment by rebstar6** [comment]
added an explicit wait on the api side, and confirmed this works, so 👍 on the code. would be great if possible to get this unit tested

File: modules/portal/public/lib/controllers/controller.records.js (hunk 1 above)
Comment (mitruly) [reply modules/portal/public/lib/controllers/controller.records.js]: I'll look into adding one now.

**Comment by mitruly** [comment]
Having issues writing a test that works as expected, as the promises in `updateRecordDisplay` are not resolving as expected. Arguably this would be testing `$q.all()`, so not sure if a new test is super necessary.

**Pull request merged**